Spatial.gd

```diff
@@ -757,6 +749,97 @@ func clear_adjustment_nodes():
+	var category: CategoryData = self.currently_active_category.get_metadata(0)
+	var icons = category.waypoint_category.get_icon()
+	icons.remove(icon_index)
+	category.category3d.remove_icon(icon_index)
+
+func set_trail_position(position: Vector3, path_index: int, point_index: int):
+	var trail_data = self.currently_active_category.get_metadata(0).category3d.paths[path_index].waypoint.get_trail_data()
+	trail_data.get_points_x()[point_index] = position.x
+	trail_data.get_points_y()[point_index] = position.y
+	trail_data.get_points_z()[point_index] = -position.z
+	refresh_path3D_points(path_index)
+	refresh_path2D_points(path_index)
+
+func reverse_trail(path_index: int):
+	var trail_data = self.currently_active_category.get_metadata(0).category3d.paths[path_index].waypoint.get_trail_data()
+	trail_data.get_points_x().invert()
+	trail_data.get_points_y().invert()
+	trail_data.get_points_z().invert()
+	refresh_path3D_points(path_index)
+	refresh_path2D_points(path_index)
+
+func get_trail_point_count(path_index: int):
+	var trail_data = self.currently_active_category.get_metadata(0).category3d.paths[path_index].waypoint.get_trail_data()
+	return len(trail_data.get_points_x())
+
+func get_trail_point_position(path_index: int, point_index: int):
+	var position: Vector3
+	var trail_data = self.currently_active_category.get_metadata(0).category3d.paths[path_index].waypoint.get_trail_data()
+	position[0] = trail_data.get_points_x()[point_index]
+	position[1] = trail_data.get_points_y()[point_index]
+	position[2] = trail_data.get_points_z()[point_index]
+	return position
+
+func add_trail_point(position: Vector3, path_index: int, point_index: int = -1):
+	var trail_data = self.currently_active_category.get_metadata(0).category3d.paths[path_index].waypoint.get_trail_data()
+	if point_index == -1:
+		trail_data.get_points_x().append(position.x)
+		trail_data.get_points_y().append(position.y)
+		trail_data.get_points_z().append(-position.z)
+	else:
+		trail_data.get_points_x().insert(point_index, position.x)
+		trail_data.get_points_y().insert(point_index, position.y)
+		trail_data.get_points_z().insert(point_index, -position.z)
+	refresh_path3D_points(path_index)
+	refresh_path2D_points(path_index)
+
+func remove_trail_point(position: Vector3, path_index: int, point_index: int = -1):
+	var trail_data = self.currently_active_category.get_metadata(0).category3d.paths[path_index].waypoint.get_trail_data()
+	trail_data.get_points_x().remove(position.x)
+	trail_data.get_points_y().remove(position.y)
+	trail_data.get_points_z().remove(-position.z)
+	refresh_path3D_points(path_index)
+	refresh_path2D_points(path_index)
+
+func new_trail_point_after(path_index: int, point_index: int):
+	var start: Vector3 = get_trail_point_position(path_index, point_index)
+	var target_position: Vector3
+	if get_trail_point_count(path_index) > point_index+1:
+		var end: Vector3 = get_trail_point_position(path_index, point_index+1)
+		target_position = ((start-end)/2) + end
```

Comment:
If this logic is now here do we still need the rout3d and rout2d functions `new_point_after()` that replicate this logic but are no longer called?

Or actually any of the self manipulation functions because now they are always rebuilt from scratch instead of manipulated.

Reply:
Lets just clean it up here, it is just a bunch of deletions of code that is being actively deprecated in this pr.